### PR TITLE
Refactor/paths from nsf

### PR DESCRIPTION
**Pull request by Gymnasiast**
This PR mostly adds object loading code. The plan to add more in follow-up PRs - most of the stuff can be backported.

### CODE REVIEW

File: src/openrct2/object/FootpathRailingsObject.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    PopulateTablesFromJson(context, root);
+}
+
+RailingEntrySupportType FootpathRailingsObject::ParseSupportType(std::string_view s)
+{
+    if (s == "pole")
+        return RailingEntrySupportType::Pole;
+    else /* if (s == "box") */
```
Comment (ZehMatt) [inline src/openrct2/object/FootpathRailingsObject.cpp:92]: I think having this explicit would be a good idea, the else should be a Guard::Assert so its impossible to have invalid inputs.

Comment (Gymnasiast) [reply src/openrct2/object/FootpathRailingsObject.cpp:92]: I don’t think it’s a good idea to have the game _crash_ on an invalid object. It should just print a warning, reject the object and move on.

Comment (ZehMatt) [reply src/openrct2/object/FootpathRailingsObject.cpp:92]: Fair enough, just wanted to highlight a potential issue here that goes easily unnoticed like all the bad objects we already have with missing images and what not.

Comment (Gymnasiast) [reply src/openrct2/object/FootpathRailingsObject.cpp:92]: I think that is mostly with .DAT objects, but you’re right, we have the opportunity to be stricter with JSON objects.

Is it okay if I create an issue for that and pick it up later? I’d need to make some changes to how the code works to properly handle this kind of thing (rejecting the object from here).

**Review by ZehMatt — APPROVED** [review]
LGTM

File: src/openrct2/object/FootpathRailingsObject.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2/object/FootpathRailingsObject.cpp:92]: Yeah sure, I would normally request changes if I feel its necessary.

**Comment by Broxzier** [comment]
This PR seems to have caused the Object Selection window to take a long time to open: https://github.com/OpenRCT2/OpenRCT2/issues/15319.

**Comment by Gymnasiast** [comment]
I’m surprised to hear that. I have no idea which of the changes could even cause such a thing.

**Comment by Broxzier** [comment]
Me neither. I've been placing breakpoints here and there, but don't get any wiser. Strangest thing is that in debug builds, it still opens instantly.

File: src/openrct2/object/Object.h [hunk 2]
```diff
@@ -42,6 +42,8 @@ enum class ObjectType : uint8_t
     TerrainEdge,
     Station,
     Music,
+    FootpathSurface,
+    FootpathRailings,
```
Comment (Broxzier) [inline src/openrct2/object/Object.h:46]: Cause found, this adds two new types, but they are not present in `object_entry_group_counts`, so it reads out of bounds. In release builds that's garbage, possibly very high numbers, hence why it takes so long.

**Comment by Gymnasiast** [comment]
Are you absolutely certain it is this commit and not one before/after?

File: src/openrct2/object/Object.h (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/object/Object.h:46]: Nice, shall I create a PR to fix?

**Pull request merged**